src/shared/Microsoft.Git.CredentialManager/IEnvironment.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System.Collections.Generic;
+
+namespace Microsoft.Git.CredentialManager
+{
+    /// <summary>
+    /// Component that encapsulates the process environment, including environment variables.
+    /// </summary>
+    public interface IEnvironment
+    {
+        /// <summary>
+        /// Current process environment variables.
+        /// </summary>
+        IReadOnlyDictionary<string, string> Variables { get; }
+
+        /// <summary>
+        /// Check if the given directory exists on the path.
+        /// </summary>
+        /// <param name="directoryPath">Path to directory to check for existence on the path.</param>
+        /// <returns>True if the directory is on the path, false otherwise.</returns>
+        bool IsDirectoryOnPath(string directoryPath);
+
+        /// <summary>
+        /// Add the directory to the path.
+        /// </summary>
+        /// <param name="directoryPath">Path to directory to add to the path.</param>
+        /// <param name="system">True to update the path for all users on the system, false to update only the user's path.</param>
+        void AddDirectoryToPath(string directoryPath, bool system);
+
+        /// <summary>
+        /// Remove the directory from the path.
+        /// </summary>
+        /// <param name="directoryPath">Path to directory to remove from the path.</param>
+        /// <param name="system">True to update the path for all users on the system, false to update only the user's path.</param>
+        void RemoveDirectoryFromPath(string directoryPath, bool system);
+
+        /// <summary>
+        /// Check if two paths are the same for the current platform and file system. Symbolic links are not followed.
+        /// </summary>
+        /// <param name="a">File path.</param>
+        /// <param name="b">File path.</param>
+        /// <returns>True if both file paths are the same, false otherwise.</returns>
+        bool IsSamePath(string a, string b);
```

Comment:
This feels a bit out of place to hang off `IEnvironment`. Should this live on `IFileSystem`?

Reply:
Yes. I'll move this to be on `IFileSystem` and the implementation be split between each platform (`WindowsFileSystem`, `Posix/MacOSFileSystem` etc)